include/tdi/common/c_frontend/tdi_info.h

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright(c) 2021 Intel Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this software except as stipulated in the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/** @file tdi_info.h
+ *
+ *  @brief Contains TDI Info APIs. Mostly to get Table and Learn Object
+ *  metadata
+ */
+#ifndef _TDI_INFO_H
+#define _TDI_INFO_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Get size of the list of TdiTable Objs
+ *
+ * @param[in] tdi_info Handle of Info object. Retrieved using
+ * tdi_info_get()
+ * @param[out] num_tables Size of array of Learn Obj pointers
+ *
+ * @return Status of the API call
+ */
+tdi_status_t tdi_num_tables_get(const tdi_info_hdl *tdi, int *num_tables);
+
+/**
+ * @brief Get Array of Table Objs from tdi_info
+ *
+ * @param[in] tdi_info Handle of Info object. Retrieved using
+ * tdi_info_get()
+ * @param[out] tdi_table_hdl_ret Array of Table Obj pointers. Memory needs
+ * to be allocated by user
+ *
+ * @return Status of the API call
+ */
+tdi_status_t tdi_tables_get(const tdi_info_hdl *tdi,
+                             const tdi_table_hdl **tdi_table_hdl_ret);
+
+/**
+ * @brief Get a Table Object from its fully qualified name
+ *
+ * @param[in] tdi_info Handle of Info object. Retrieved using
+ * tdi_info_get()
+ * @param[in] table_name Fully qualified P4 Table Obj name
```

Comment:
Aren't the tables TDI generic and not P4 specific?? If TDI generic then remove P4